src/imputation/imputation_helpers.py

```diff
@@ -262,10 +262,11 @@ def split_df_on_trim(df: pd.DataFrame, trim_bool_col: str) -> pd.DataFrame:
 
     if not df.empty:
         # TODO: remove this temporary fix to cast Nans to False
-        df[trim_bool_col].fillna(False, inplace=True)
+        df_copy = df.copy()
+        df_copy.loc[:, trim_bool_col].fillna(False, inplace=True)
```

Comment:
I'm never sure when it's necessary to use .copy(), appart from to get rid of the warnings! 

For this pipeline, we prefer to avoid the inplace=True and explicitly set the updated column, though I can see it was used here.

What about this:
```suggestion
        df[trim_bool_col] = df.copy()[trim_bool_col].fillna(False)
   # or, if it still complains
        df.loc[:,trim_bool_col] = df.copy().loc[:,trim_bool_col].fillna(False)
```

Reply:
Noted on the avoidance of `inplace=True` for this pipeline. The implementation of `inplace=True` for this line was implemented a previous developer and was approved November last year, therefore I assumed its used was tolerated. 

The first suggestion causes the pipeline to crash and the second suggestion also kills the pipeline. I have modified the line from this:

```Python
df_copy.loc[:, trim_bool_col].fillna(False, inplace=True)
```
to:
``` Python
df_copy.loc[:, trim_bool_col] = df_copy.loc[:, trim_bool_col].fillna(False)
```

The pipeline works with this latest version and satisfies the removal of `inplace=True`.

Reply:
thanks for checking all those things.